app/javascript/lib/validation/validate-remote-file-content.js

```diff
@@ -0,0 +1,77 @@
+/**
+* @fileoverview Client-side file validation (CSFV) for sync
+*/
+
+import { validateFileContent } from './validate-file-content'
+import { fetchBucketFile } from 'lib/scp-api'
+
+/**
+ * 50 MiB, max number of bytes to read and validate from remote file for sync.
+ * Median fixed US bandwidth is 16 MiB/s (17 MB/s, 136 Mbps) as of 2021-12
+ * per https://www.speedtest.net/global-index/united-states.  > 80% of ingests
+ * are for files <= 50 MiB per https://mixpanel.com/s/3EbFMj.
+ *
+ * So 50 MiB means sync CSFV fully scans > 80% files, usually in < 5 seconds.
+ * Local tests gave 4.6 s (3.4 s remote read, 1.1 s validate; 138 Mbps down).
+ */
+export const MAX_SYNC_CSFV_BYTES = 50 * 1024 * 1024
+
+/** Add file-size-related entries to syncProps */
+export function updateSyncProps(contentRange, contentLength, syncProps) {
+  // Total size of the file in bytes
+  const fileSizeTotal = parseInt(contentRange.split('/')[1])
+
+  // Total bytes downloaded, which can be much less than total file size
+  const fileSizeFetched = parseInt(contentLength)
+
+  const fetchedCompleteFile = (fileSizeTotal === fileSizeFetched)
+  Object.assign(syncProps, {
+    fileSizeFetched,
+    fileSizeTotal,
+    fetchedCompleteFile
+  })
+
+  return syncProps
+}
+
+/** Construct a File object from superficially-parsed range response data */
+export function getFileFromRangeData(content, fileName, contentType, syncProps) {
+  // If Range request didn't fetch the full file, then truncate the last
+  // line, which is almost certainly incomplete and thus invalid.
+  let cleanLineContent = content
+  if (!syncProps.fetchedCompleteFile) {
+    cleanLineContent = content.split('\n').slice(0, -1).join('\n')
```

Comment:
Done, including an optimization to check on each chunk instead of each line.

Out of curiosity, afterwards I benchmarked the original performance:

```
const newLineRegex = /\r?\n/
const text60MB =('foo'.repeat(100) + '\n').repeat(200_000)
const numBytes = (new TextEncoder().encode(text60MB)).length
console.log('Bytes in string: ' + numBytes)
const t0 = Date.now()
const cleanText = text60MB.split(newLineRegex).slice(0, -1).join(newLineRegex)
const t1 = Date.now()
console.log('Milliseconds to split and rejoin 60 MB: ' + (t1 - t0))

Bytes in string: 60200000
Milliseconds to split and rejoin 60 MB: 113
```

So the original way here is at worst ~2% slower (113 ms/5000 ms vs. ?/5000), but less complex and more self-contained (across 2 simple functions in 1 module vs. across 3 functions, one quite complex, in 3 modules).  If the update per suggestion is problematic I'd go with the original approach, but export `newLineRegex` from over there and use it here.

I also noticed (after refactoring, of course!) that the ticket notes refining this to use ChunkedLineReader is out of scope.